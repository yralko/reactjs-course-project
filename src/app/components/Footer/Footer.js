import React from 'react';
import Logo from '../Logo/Logo';
import classes from './index.css';

const Footer = () => (
  <footer className={classes.Footer}>
    <div className={classes.wrapper}>
      <Logo />
    </div>
  </footer>
);

export default Footer;
