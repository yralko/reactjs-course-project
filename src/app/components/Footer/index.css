.Footer {
  height: 70px;
  background-color: #414141;
}

.Footer .wrapper {
  width: 70%;
  margin: 0 auto;
}
