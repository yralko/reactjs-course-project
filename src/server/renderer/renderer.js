import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter as Router } from 'react-router-dom';
import App from '../../app/App';


export default (req, store, context) => {
  const root = (
    <Provider store={store}>
      <Router location={req.url} context={context}>
        <App />
      </Router>
    </Provider>
  );

  const content = renderToString(root); 

  return `
  <!doctype html>
  <html lang="en">
  <head>
    <meta charset="utf-8">
    <title>The HTML5 Herald</title>
    <base href="/" />
    <link rel="stylesheet" href="style.css" />
  </head>
  <body>
    <div id="root">${content}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState())}</script>
    <script src="bundle.js"></script>
  </body>
  </html>
  `;
};
