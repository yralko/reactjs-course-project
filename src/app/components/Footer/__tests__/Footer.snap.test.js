import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

test('matches snapshot', () => {
  const tree = renderer.create(<Footer />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
