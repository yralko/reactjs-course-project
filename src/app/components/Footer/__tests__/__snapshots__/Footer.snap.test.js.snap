// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<footer
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <span>
        netflixroulette
      </span>
    </div>
  </div>
</footer>
`;
